js/bootstrap-datepicker.js

```diff
@@ -1196,6 +1200,10 @@
 			}
 
 			if (!target.hasClass('disabled')){
+				// Allow clicking on elements inside a day
+				if (target.parent().hasClass('day'))
+			        target = target.parent();
```

Comment:
I'm not sure that e.currentTarget would work because bubbling/propagation are stopped at the top of the click handler.

Instead, I use .parents(), does that work?